Zend/zend_object_handlers.c

```diff
@@ -1512,6 +1512,15 @@ ZEND_API zval *zend_std_get_static_property_with_info(zend_class_entry *ce, zend
 	ret = CE_STATIC_MEMBERS(ce) + property_info->offset;
 	ZVAL_DEINDIRECT(ret);
 
+	/* TODO: Always make this an error, not just for typed properties. */
+	if (UNEXPECTED((ce->ce_flags & ZEND_ACC_TRAIT) && ZEND_TYPE_IS_SET(property_info->type))) {
+		zend_throw_error(NULL,
+			"Cannot access typed static property %s::$%s on a trait, "
+			"it may only be used as part of a class",
```

Comment:
Style nit: can we please not break up the string literal parts? It makes it hard to grep for error messages.